Source/Dafny/DafnyOptions.cs

```diff
@@ -17,6 +17,12 @@ namespace Microsoft.Dafny {
   public class DafnyOptions : Bpl.CommandLineOptions {
     private ErrorReporter errorReporter;
 
+    public new static DafnyOptions FromArguments(params string[] arguments) {
```

Comment:
While looking at the tests, the call `DafnyOptions.FromArguments()` felt odd to me, because I was looking for "where are the static/global arguments defined that this function pulls from?". I did not realize immediately that it's the empty parameter list that was the arguments, until I came back to this function.

I don't have a strong opinion on that (consider it optional), but I would have felt much more comfortable with a name such as `Create` so that  `DafnyOption.Create()` really feels like it's not depending on any external context. It would still feel natural that: `DafnyOptions.Create("--compile:0"). is actually behaving like command-line arguments.